mimic_joint/proposal.md

```diff
@@ -0,0 +1,175 @@
+
+For proposals with many "Proposed changes":
+
+* Number each group of like changes under a descriptive `###` subheading.
+* Individual changes under `####` subheadings.
+* "Alternatives considered" for each individual change under `#####` subheading.
+* If "Previous behavior" and/or "Justification" sections are shared by all the
+individual changes under one `###` heading, those parts can go directly under
+the `###` heading instead of under each `####` heading.
+
+## Examples
+
+There is an example `gearbox` joint in the
+[demo\_joint\_types](https://github.com/osrf/gazebo_models/blob/master/demo_joint_types/model.sdf#L156-L328)
+model, consisting of three links,
+
+~~~
+        <link name="gearbox_base">
+            <pose >-.49 0 0.35 0 0 0</pose>
+            <!-- ... -->
+        </link>
+        <link name="gearbox_input">
+            <pose >-.38 -0.075 0.55 0 0 0</pose>
+            <!-- ... -->
+            <visual name="gearbox_input_visual">
+                <geometry>
+                    <box>
+                        <size>0.1 0.25 0.1</size>
+                    </box>
+                </geometry>
+            </visual>
+        </link>
+        <link name="gearbox_output">
+            <pose >-.3 0.0 0.55 0 1.5708 0</pose>
+            <visual name="gearbox_output_visual">
+                <geometry>
+                    <cylinder>
+                        <radius>0.1</radius>
+                        <length>0.05</length>
+                    </cylinder>
+                </geometry>
+            </visual>
+        </link>
+~~~
+
+two `revolute` joints,
+
+~~~
+        <!-- Gearbox links revolute joints, so create a couple revolute joints -->
+        <joint name="gearbox_input_joint" type="revolute">
+            <parent>gearbox_base</parent>
+            <child>gearbox_input</child>
+            <axis>
+                <xyz>1 0 0</xyz>
+            </axis>
+            <pose>0 0.075 0 0 0 0</pose>
+        </joint>
+        <joint name="gearbox_output_joint" type="revolute">
+            <parent>gearbox_base</parent>
+            <child>gearbox_output</child>
+            <axis>
+                <xyz>0 0 1</xyz>
+            </axis>
+            <pose>0 0 0 0 0 0</pose>
+        </joint>
+~~~
+
+and one `gearbox` joint.
+
+~~~
+        <joint name="gearbox_demo" type="gearbox">
+            <parent>gearbox_input</parent>
+            <child>gearbox_output</child>
+            <gearbox_reference_body>gearbox_base</gearbox_reference_body>
+            <gearbox_ratio>5</gearbox_ratio>
+            <!-- input axis (relative to child) -->
+            <axis>
+                <xyz>0 0 1</xyz>
+            </axis>
+            <!-- output axis (relative to child) -->
+            <axis2>
+                <xyz>0 0 1</xyz>
+            </axis2>
+        </joint>
+~~~
+
+The `gearbox` joint could be equivalently and more concisely expressed as
+the following `mimic` joint:
+
+~~~
+        <joint name="mimic_demo" type="mimic">
```

Comment:
actually, URDF has a `<mimic />` tag that can apply to any `<joint />` instead of a `type`; let me think about that

Reply:
https://github.com/ros/urdfdom/blob/36be7f8f12c6269483b9f0d9c85b5a92a1c9e7ad/urdf_parser/test/urdf_unit_test.cpp#L184

Reply:
I think I would be inclined to add `//joint/axis/mimic` and `//joint/axis2/mimic` tags instead of adding a new joint type. Let's update the proposal accordingly

Reply:
Edit : https://github.com/gazebosim/sdf_tutorials/pull/80